### PR TITLE
Implement light channels

### PR DESCRIPTION
**Pull request by pixelflinger**
We use fp16 for most values (e.g.: direction, color), which 
ultimately frees a vec4 (Currently unused) and a few other bits.

### CODE REVIEW

File: libs/filamat/src/shaders/CodeGenerator.cpp [hunk 1]
```diff
@@ -111,6 +111,30 @@ io::sstream& CodeGenerator::generateProlog(io::sstream& out, ShaderType type,
...
+float u16tofp32(highp uint v) {
+    // this doesn't handle denormals, NaNs or inf
+    v <<= 16u;
+    highp uint s = v & 0x80000000u;
+    highp uint n = v & 0x7FFFFFFFu;
+    highp uint nz = n == 0u ? 0u : 0xFFFFFFFF;
+    return uintBitsToFloat(s | ((((n >> 3u) + (0x70u << 23))) & nz));
+}
+
+vec2 unpackHalf2x16(highp uint v) {
```
Comment (romainguy) [inline libs/filamat/src/shaders/CodeGenerator.cpp:131]: Do we expect naming conflicts depending on the GL version? Should we use a different name?

**Review by romainguy — APPROVED** [review]

File: libs/filamat/src/shaders/CodeGenerator.cpp (hunk 1 above)
Comment (pixelflinger) [reply libs/filamat/src/shaders/CodeGenerator.cpp:131]: Yeah, I'm afraid of that. That's why I need to try on a windows or Linux with at least 4.2.

**Review by romainguy — APPROVED** [review]

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ A new header is inserted each time a *tag* is created.
 
 - engine: Option to automatically compute bent normals from SSAO & apply to specular AO
   [⚠️ **Material breakage**].
+- engine: New APIs: Light channels. Geometry and lights now have a channel associated to them, at 
+  least one channel must match for lighting to occur [⚠️ **Material breakage**].
```
Comment (bejado) [inline RELEASE_NOTES.md:11]: Should we bump this release to v1.12.0?

File: shaders/src/getters.vs [hunk 3]
```diff
@@ -69,14 +69,14 @@ vec4 getPosition() {
 
 #if defined(HAS_SKINNING_OR_MORPHING)
 
-    if (objectUniforms.morphingEnabled == 1) {
+    if ((objectUniforms.flags & 0x2u) != 0u) {
```
Comment (bejado) [inline shaders/src/getters.vs:72]: Make some defines for clarity?
```
#define SKINNING_ENABLED_BIT 0x1u
#define MORPHING_ENABLED_BIT 0x2u
#define CONTACT_SHADOWS_BIT 0x4u
```

File: RELEASE_NOTES.md (hunk 2 above)
Comment (pixelflinger) [reply RELEASE_NOTES.md:11]: I think so.

File: shaders/src/getters.vs (hunk 3 above)
Comment (pixelflinger) [reply shaders/src/getters.vs:72]: Now that's a good idea!

File: RELEASE_NOTES.md (hunk 2 above)
Comment (romainguy) [reply RELEASE_NOTES.md:11]: Yes definitely.

**Pull request merged**